Provide the user the option to use GPG to sign packages in sandbox

drscream opened the pull request:
This should fix the issue #6 and create entries in the `pkg_install.conf` file if needed and if the user configure the `PKGSRC_GPG_SIGN_AS` variable.

jperkin commented:
Thanks for this! Could you add a check to only add them if they do not already exist? While it shouldn't hurt to have multiple entries for users like myself who already have GPG configured in `pkg_install.conf` I'd prefer to avoid any corner cases.

drscream commented:
Sure! I thought `run-sandbox` only creates new sandboxes and will remove them after logout so no duplicate entires should be possible.

jperkin commented:
It does, but the pbulk bootstraps that we use have them already baked into `pkg_install.conf` (same for anyone who builds their own bootstraps with the GPG stuff set).  The public bootstraps are generated from our internal bootstraps but with the signing stuff removed.

jperkin commented:
https://github.com/joyent/pkgbuild/blob/master/scripts/mk-pbulk-bootstrap#L57..L66

drscream commented:
Thanks for the link, I've added an additional commit with the `ed` command to delete the lines if they exist already. I tried a little bit longer todo the insert also with `ed` and remove the `cat` but it looks like I'm not so talented to use `ed` :-)

Anyway I've added the remove `ed` part to the `if` to be sure we will only remove it if the `PKGSRC_GPG_SIGN_AS` is used.

jperkin commented:
I've tidied it a little and ensured that for users who have already set them there will be no changes to the generated file. Thanks for the patch!
